image.go

```diff
@@ -47,9 +56,62 @@ type ImageConfig struct {
 	User string `yaml:"user,omitempty" json:"user,omitempty"`
 }
 
-// MergeImageConfig copies the fields from the source [ImageConfig] into the destination [dalec.DockerImageSpec].
+type imageBuilderConfig struct {
+	platform *ocispecs.Platform
+}
+
+type ConfigOpt func(*imageBuilderConfig)
+
+func WithPlatform(p ocispecs.Platform) ConfigOpt {
+	return func(c *imageBuilderConfig) {
+		c.platform = &p
+	}
+}
+
+func BuildImageConfig(ctx context.Context, client gwclient.Client, spec *Spec, targetKey string, dflt string, opts ...ConfigOpt) (*DockerImageSpec, error) {
```

Comment:
We should not be importing the gateway client in this package.
Instead of doing the `ResolveImageConfig` in here, can callers just pass an image config?

That also makes this more easily testable.

Reply:
Sure